src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php

```diff
@@ -217,23 +217,40 @@ public function regenerate($destroy = false, $lifetime = null)
      */
     public function save()
     {
+        $session = $_SESSION;
+
+        $bags = array_merge($this->bags, array($this->metadataBag));
+
+        foreach ($bags as $bag) {
+            if (empty($_SESSION[$key = $bag->getStorageKey()])) {
```

Comment:
This is a good try, but will not work because the metadataBag is never empty. Once the session is started, the metadataBag is created ans filled with some metedata that always will be there as long as the session lives.

Another concern I have is that this soluton is less general, and every new type of session storage has to be changed to make sure it stores nothing in the session when the bags are empty. This certainly will be a bigger BC break for those that write their own sessionStorage class, than just adding a "wasStarted" or "isActive" method.

Reply:
Fixed, thanks. The approach in this PR fixes more issues with sessions, and works fine when `$_SESSION` is used directly, so it definitely the way to go to me.